File: app/src/main/java/com/datarockets/mnchkn/activities/players/PlayersListPresenterImpl.java
package com.datarockets.mnchkn.activities.players;

import android.content.Context;

import com.datarockets.mnchkn.models.Player;

import java.util.List;

public class PlayersListPresenterImpl implements PlayersListPresenter,
        PlayersListInteractor.OnFinishedListener {

    private PlayersListView mPlayersListView;
    private PlayersListInteractor mInteractor;
    private Context mContext;

    public PlayersListPresenterImpl(PlayersListView playersListView, Context context) {
        this.mContext = context;
        this.mPlayersListView = playersListView;
        this.mInteractor = new PlayersListInteractorImpl(context);
    }

    @Override
    public void checkIsEnoughPlayers() {
        if (mPlayersListView != null) {
            mInteractor.checkIsEnoughPlayer(this);
        }
    }

    @Override
    public void addPlayer(String name) {
        if (mPlayersListView != null) {
            mInteractor.addPlayer(name, this);
        }
    }

    @Override
    public void deletePlayerListItem(int position, long id) {
        if (mPlayersListView != null) {
            mInteractor.deletePlayer(position, id, this);
        }
    }

    @Override
    public void clearPlayersStats() {
        if (mPlayersListView != null) {
            mInteractor.clearPlayersStats(this);
        }
    }

    @Override
    public void clearGameSteps() {
        if (mPlayersListView != null) {
            mInteractor.clearGameSteps();
        }
    }

    @Override
    public void setGameStarted() {
        if (mPlayersListView != null) {
            mInteractor.setGameStatus(true);
        }
    }

    @Override
    public void setGameFinished() {
        if (mPlayersListView != null) {
            mInteractor.setGameStatus(false);
        }
    }

    @Override
    public void onCreate() {
        if (mPlayersListView != null) {
            mInteractor.isGameStarted(mContext, this);
        }
    }

    @Override
    public void onResume() {
        if (mPlayersListView != null) {
            mInteractor.getPlayers(this);
        }
    }

    @Override
    public void onDestroy() {
        if (mPlayersListView != null) {
            mPlayersListView = null;
        }
    }

    @Override
    public void onPlayersLoaded(List<Player> players) {
        if (mPlayersListView != null) {
            mPlayersListView.setPlayersList(players);
        }
    }

    @Override
    public void onPlayerAdded(Player player) {
        if (mPlayersListView != null) {
            mPlayersListView.addPlayerToList(player);
        }
    }

    @Override
    public void onPlayerDeleted(int position) {
        if (mPlayersListView != null) {
            mPlayersListView.deletePlayerFromList(position);
        }
    }

    @Override
    public void onPlayersCountChecked(boolean enough) {
        if (mPlayersListView != null) {
            if (enough) {
                mPlayersListView.launchDashboard();
            } else {
                mPlayersListView.showWarning();
            }
        }
    }

    @Override
    public void onGameStarted(boolean started) {
        if (mPlayersListView != null && started) {
            mPlayersListView.showStartContinueDialog();
        }
    }

}

File: app/src/main/kotlin/com/datarockets/mnchkn/activities/players/PlayersListPresenterImpl.kt
package com.datarockets.mnchkn.activities.players

import com.datarockets.mnchkn.models.Player

class PlayersListPresenterImpl(private var mPlayersListView: PlayersListView?,
                               private val mInteractor: PlayersListInteractor) : PlayersListPresenter, PlayersListInteractor.OnFinishedListener {

    override fun checkIsEnoughPlayers() {
        if (mPlayersListView != null) {
            mInteractor.checkIsEnoughPlayer(this)
        }
    }

    override fun addPlayer(name: String) {
        if (mPlayersListView != null) {
            mInteractor.addPlayer(name, this)
        }
    }

    override fun deletePlayerListItem(position: Int, id: Long) {
        if (mPlayersListView != null) {
            mInteractor.deletePlayer(position, id, this)
        }
    }

    override fun clearPlayersStats() {
        if (mPlayersListView != null) {
            mInteractor.clearPlayersStats(this)
        }
    }

    override fun clearGameSteps() {
        if (mPlayersListView != null) {
            mInteractor.clearGameSteps()
        }
    }

    override fun setGameStarted() {
        if (mPlayersListView != null) {
            mInteractor.setGameStatus(true)
        }
    }

    override fun setGameFinished() {
        if (mPlayersListView != null) {
            mInteractor.setGameStatus(false)
        }
    }

    override fun onCreate() {
        if (mPlayersListView != null) {
            mInteractor.isGameStarted(this)
        }
    }

    override fun onResume() {
        if (mPlayersListView != null) {
            mInteractor.getPlayers(this)
        }
    }

    override fun onDestroy() {
        if (mPlayersListView != null) {
            mPlayersListView = null
        }
    }

    override fun onPlayersLoaded(players: List<Player>) {
        if (mPlayersListView != null) {
            mPlayersListView!!.setPlayersList(players)
        }
    }

    override fun onPlayerAdded(player: Player) {
        if (mPlayersListView != null) {
            mPlayersListView!!.addPlayerToList(player)
        }
    }

    override fun onPlayerDeleted(position: Int) {
        if (mPlayersListView != null) {
            mPlayersListView!!.deletePlayerFromList(position)
        }
    }

    override fun onPlayersCountChecked(enough: Boolean) {
        if (mPlayersListView != null) {
            if (enough) {
                mPlayersListView!!.launchDashboard()
            } else {
                mPlayersListView!!.showWarning()
            }
        }
    }

    override fun onGameStarted(started: Boolean) {
        if (mPlayersListView != null && started) {
            mPlayersListView!!.showStartContinueDialog()
        }
    }

}